core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/models.scala

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.dsde.workbench.leonardo
+
+import ca.mrvisser.sealerate
+import org.broadinstitute.dsde.workbench.model.google.GoogleProject
+
+final case class WorkspaceId(value: String) extends AnyVal
+
+final case class CloudContextDb(value: String) extends AnyVal
+
+sealed abstract class CloudContext extends Product with Serializable {
+  def asString: String
+  def asStringWithProvider: String
+  def cloudProvider: CloudProvider
+  def asCloudContextDb: CloudContextDb = CloudContextDb(asString)
+}
+object CloudContext {
+  final case class Gcp(value: GoogleProject) extends CloudContext {
+    override val asString = value.value
+    override val asStringWithProvider = s"Gcp/${value.value}"
+    override def cloudProvider: CloudProvider = CloudProvider.Gcp
+  }
+  final case class Azure(value: ManagedResourceGroup) extends CloudContext {
```

Comment:
Similar to above comment, wondering if `CloudContext.Azure` should have `(tenantId: TenantId, subscriptionId: SubscriptionId, mrg: ManagedResourceGroupName)`